### ISSUE TITLE
another crash :) 

### ISSUE THREAD

**Issue by Dmorgen** [smith-chem-wisc/MetaMorpheus#1024]
[results.txt](https://github.com/smith-chem-wisc/MetaMorpheus/files/1799380/results.txt)
ran GPTMD first, then search when it has failed. 
with this search I've added a few more modifications (sample artefacts) - which might be part of the protblem...

Thanks!
David.

**Comment by rmillikin** [comment]
Thanks for the crash report. It looks like your system ran out of memory when generating the peptide index in modern search. Can you try increasing the number of database partitions? This should split the database into pieces, making it easier for your computer to handle. Alternatively I suppose you could use a higher-RAM machine. Can you try these and report back?

**Comment by Dmorgen** [comment]
Will try to divide more – I’ve used 20 partitions on a 128GB VM for this search.

**Comment by rmillikin** [comment]
Ok, great. I will check to make sure we're partitioning the database correctly. I'm also going to add a more meaningful error for this crash.

Can you send me your toml files and your database? This will help me debug the issue.

**Comment by Dmorgen** [comment]
will, kinda lost connection with the server currently.

**Comment by Dmorgen** [comment]
please use the following link:
https://owncloud.incpm.weizmann.ac.il/index.php/s/WEvhFnRh98uMbdL

**Comment by rmillikin** [comment]
Ok, I think I see the issue. The number of variable mods in your search is likely the problem. I'm seeing 11 variable mods, we recommend 1, maybe 2 at most. GPTMD is much better at finding mods in terms of FDR, search-time, and computational efficiency. Even if you got this search running it would probably take days to run.

So, I recommend re-doing your GPTMD task with all of these mods checked (except oxidation of M). They will be annotated in the XML database. Then, do your search task with only oxidation of M as variable mod.

**Comment by Dmorgen** [comment]
will do, thanks!

**Comment by rmillikin** [comment]
Did that strategy work out for you? Any more crashes?

Rob

**Comment by Dmorgen** [comment]
no - it runs well currently, but only in classic mode (conservation)

**Comment by rmillikin** [comment]
Are you still getting crashes with Modern Search? And if so, are they "out of memory" type crashes in particular?

**Comment by Dmorgen** [comment]
yes. the memory maxes out and the computer freezes, then crashes.

**Comment by rmillikin** [comment]
Can you send me your .toml files and your database for that search? If it's crashing on your 128gb machine we should investigate the cause of this.

**Comment by Dmorgen** [comment]
cheers,
D.

[MM_bugs.zip](https://github.com/smith-chem-wisc/MetaMorpheus/files/1819548/MM_bugs.zip)

**Comment by rmillikin** [comment]
This actually isn't a memory issue, it's an old bug in modern search; I fixed it in v0.0.265. Please try updating to the latest version of MM (0.0.272) and give that a shot.

**Comment by Dmorgen** [comment]
I am running .272...

**Comment by Dmorgen** [comment]
here is another, from another machine (added the .txt so i could attach it).
[GptmdTaskconfig.toml.txt](https://github.com/smith-chem-wisc/MetaMorpheus/files/1819601/GptmdTaskconfig.toml.txt)

**Comment by rmillikin** [comment]
At the top of the results file you sent it says "MetaMorpheus: version 0.0.260.0"; can you check that that crash result.txt is the correct one?

I see another issue which is a problem with our GUI being unclear. The "MaxDegreeOfParallelism" is the number of files to analyze in parallel. If you put 24 here, MM will attempt to load in 24 files at once and search all of them simultaneously. I usually use 1 or 2 but my machine is not as powerful as yours; you may be able to go up to 8 or 10. The single-file search itself is parallelized and uses the maximum threads available, so even if you put a 1 for MaxDegreeOfParallelism it will still parallelize the search (this is very confusing). I will update the GUI to be more clear.

**Comment by Dmorgen** [comment]
Ok, thanks. the version thing is confusing - I think I've upgraded them both just as .272 got out, but maybe I'm wrong. I'll try running a modern search on one machine when it is done running the current search, just to see how it goes. 

I've got a 26 core VM running the searches - does it use them all even with 6-10 degrees of parallelism?

**Comment by rmillikin** [comment]
Ok, sounds good about the versions.

Some parts of the search are parallelized and some parts are not; for the parts that are parallizable, MM will use all available threads in parallel even if you put 1 for the degrees of parallelism. The current "max degrees of parallelism" option is just going to be renamed to something like "max number of files to analyze in parallel".

**Comment by Dmorgen** [comment]
another day, another crash :)
this one finished the GPTMD run and crashed on search. 

the next one is now stuck on GPTMD
[GptmdTaskconfig - Copy2.toml.txt](https://github.com/smith-chem-wisc/MetaMorpheus/files/1824279/GptmdTaskconfig.-.Copy2.toml.txt)


[SearchTaskconfig - Copy.toml.txt](https://github.com/smith-chem-wisc/MetaMorpheus/files/1824272/SearchTaskconfig.-.Copy.toml.txt)
[results.txt](https://github.com/smith-chem-wisc/MetaMorpheus/files/1824275/results.txt)
[prose.txt](https://github.com/smith-chem-wisc/MetaMorpheus/files/1824276/prose.txt)

**Comment by zrolfs** [comment]
Thanks for your patience with all of the crashes. The results file you posted appears to be the results from GPTMD, which didn't crash. Can you post the results from the search that crashed? I am suspicious that it is still a semi-trypsin problem.

**Comment by Dmorgen** [comment]
So the first Toml file is from a run which stalled on memory during GPTMD. 
the last 3 files belong to a seperate search task (the 2nd Toml) that stalled after successfully runing a GPTMD task (the last 2 files). does it make sense?

**Comment by zrolfs** [comment]
That makes sense to me, but the results.txt appears to be from a gptmd task (not a search task) and doesn't indicate that there were any problems. This is the results.txt that was in the same folder as the SearchTaskconfig, correct? Did the search task stall (high RAM or endless loop) or crash (warning message?). Separately, when you ran GPTMD again, did it stall because of RAM or something else? Was there anything different between the first and second GPTMD?

**Comment by Dmorgen** [comment]
There weren't any results file written - the searches did stall, froze the computer due to high RAM. got "Not responding" status in task manager. i think the 2nd GPTMD was done on tryptic alone, while the first was semi-tryptic.

**Comment by zrolfs** [comment]
Thanks, that's good to know. Can you double check the GPTMD settings? The second one is semi-tryptic (I can see in the toml), so the first was probably tryptic.

I'll continue looking into this and keep you posted on the resolution.
In the meantime, tryptic seaches (not semi-tryptic) should be functional.

**Comment by Dmorgen** [comment]
here's the successful GPTMD search.
[GptmdTaskconfig - Copy.toml.txt](https://github.com/smith-chem-wisc/MetaMorpheus/files/1825485/GptmdTaskconfig.-.Copy.toml.txt)

**Comment by rmillikin** [comment]
as noted in #1042 , try setting the "max modification isoforms" to something like 20 to resolve some RAM issues. This problem seems to be especially prevalent in the search following GPTMD (lots of PTMs annotated, lots of isoforms possible). We're not quite sure what the repercussions of setting this value lower are yet.

**Comment by rmillikin** [comment]
Please try the new version (0.0.274). Even with mod isoforms set to 4096, RAM usage in classic search/GPTMD is far lower (~2.5 GB instead of 36 GB in my tests). Those types of searches also should be faster. Modern Search and NonSpecific search haven't changed.

I'm going to go ahead and close this issue; feel free to open a new one if/when you encounter new problems.
